include/pybind11/cast.h

```diff
@@ -34,6 +34,20 @@ template <typename Type> class opaque {
 
 NAMESPACE_BEGIN(detail)
 
+/// Convert from any type pointer to another avoding warnings about void* vs (void*()) incompatibility.
```

Comment:
Can you give this a try? I.e. add 
``
#if __GNUC_PREREQ(4,8)
#  pragma GCC diagnostic ignored "-Wpendantic"
#endif
``
to the files in question?